modules/oe_webtools_analytics/modules/oe_webtools_analytics_rules/src/EventSubscriber/WebtoolsAnalyticsEventSubscriber.php

```diff
@@ -71,9 +83,15 @@ public function setSection(AnalyticsEventInterface $event): void {
     $rules = $storage->loadMultiple();
     $current_uri = $this->requestStack->getCurrentRequest()->getRequestUri();
     /** @var \Drupal\oe_webtools_analytics_rules\Entity\WebtoolsAnalyticsRuleInterface $rule */
+    if ($cached_section = $this->cache->get($current_uri)) {
```

Comment:
I discussed this with @imanoleguskiza to potentially store some more info in this cache in the future. For now we have the section but later we may have something else that we need to put on the event.

So let's have the cache data an array with one key called `section` whose value is the section. Otherwise the entire cache backend is also just for analytics sections only which doesn't make sense.

Reply:
Changes and improvements is applied

Reply:
Hi @sergepavle , it doesn't seem that you took @upchuk's considerations into account. What he is requesting is that we allow this cache to be used in more contexts than just the section rules.
For each cache item id (the uri or path), we need to be able to cache more than one parameter. In the current state, we only give the option to store the section. So the end result of the cache should be $cache[$cacheID]['section'][$section], so in the future we can have, for example, $cache[$cacheID]['instance][$instance] (instance being a parameter provided in another place.